common4j/src/main/com/microsoft/identity/common/java/util/ported/LocalBroadcaster.java

```diff
@@ -0,0 +1,54 @@
+//  Copyright (c) Microsoft Corporation.
+//  All rights reserved.
+//
+//  This code is licensed under the MIT License.
+//
+//  Permission is hereby granted, free of charge, to any person obtaining a copy
+//  of this software and associated documentation files(the "Software"), to deal
+//  in the Software without restriction, including without limitation the rights
+//  to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+//  copies of the Software, and to permit persons to whom the Software is
+//  furnished to do so, subject to the following conditions :
+//
+//  The above copyright notice and this permission notice shall be included in
+//  all copies or substantial portions of the Software.
+//
+//  THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+//  IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+//  FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+//  AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+//  LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+//  OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+//  THE SOFTWARE.
+package com.microsoft.identity.common.java.util.ported;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import lombok.NonNull;
+
+public enum  LocalBroadcaster {
+    INSTANCE;
+
+    public interface IReceiverCallback {
+        void onReceive(@NonNull final DataBag dataBag);
+    }
+
+    final Map<String, IReceiverCallback> mReceivers = new HashMap<>();
+
+    public void registerCallback(@NonNull final String alias, @NonNull final IReceiverCallback callback){
```

Comment:
Do we want to let this stomp on existing values?

Reply:
This is OK, since it is only used on interactive flows, and those run in a single thread right now.  As a safety feature, you could throw an exception if an overwrite was attempted and then have the register methods that are run enclosed in a try-finally to make sure listeners release these.